### ISSUE TITLE
contrast_logits may be negative?

### ISSUE THREAD

**Issue by ran337287** [liyi01827/noisy-contrastive#1]
https://github.com/liyi01827/noisy-contrastive/blob/03f48ca3028f4a6fc54b6faaf05ad811b4055381/main.py#L216

hi, I have some question, about contrast_logits=2+contrast_1. What is the meaning of 2?
contranst_1= -contrast_1 * torch.zeros(bsz, bsz).fill_diagonal_(1).type_as(outputs) + (
                     (1 - contrast_1).log()) * torch.ones(bsz, bsz).fill_diagonal_(0).type_as(outputs)
means the similarity of <pi, pi> and (1-sim).log() for <pi, pj> or <pj, pi>.
but 2+contrast_1 cannot make the contrast_logits be positive (<pi, pi> is 0~1, but (1-sim).log() is (-inf, 0)).

**Comment by ran337287** [comment]
p1=torch.clamp(p1, 1e-4, 1.0-1e-4) should be adapted for dim?

**Comment by liyi01827** [comment]
Hi, "2" here follows the [BYOL](https://arxiv.org/pdf/2006.07733.pdf) loss (Eq.2). If you are using [SimSiam](https://arxiv.org/pdf/2011.10566.pdf) loss (Eq.1), then there is no "2". Basically, the loss functions are set to be consistent with previous work. We don't intend to make contrast_logits positive. 

Thanks for your question.
